File: official/convex.workspace.runtime/implementations/convex_interlocking_infrastructure/fixtures/dirty/interlocking_resolution_model/convex/trains/interlocking.ts
// DIRTY fixture — coder.convex.interlocking-resolution-model-exists.
// resolveTrain resolves a BARE trainId string; there is no structured InterlockingResolution model
// carrying routeId/guardId/category/... Expected: >=1 violation (bare-train-id-resolution).
import { TrainRunner } from "./runner";

export class InterlockingRunner {
  private readonly path: string;
  constructor(interlockingYamlPath: string) {
    this.path = interlockingYamlPath;
  }

  // BUG: returns a bare trainId string — loses routeId, guardId, category, categoryDigit, reason, ...
  resolveTrain(action: string, inputs: Record<string, unknown>): string {
    return "3001-solo-match-complete";
  }

  execute(action: string, inputs: Record<string, unknown>) {
    const trainId = this.resolveTrain(action, inputs);
    return new TrainRunner(`plan/_trains/${trainId}.yaml`).execute(trainId, inputs);
  }
}

File: official/atdd.extension.coder.train-interlocking/implementations/interlocking-infrastructure/fixtures/dirty_bare_resolution/python/trains/interlocking.py
"""InterlockingRunner — DIRTY (bare resolution) fixture.

Runner + resolve_train exist and the Station Master is wired, but resolve_train returns a bare
train_id string and NO structured InterlockingResolution model is defined. Only
coder.train.interlocking-resolution-model-exists should fire.
"""
from trains.runner import TrainRunner


class InterlockingRunner:
    def __init__(self, interlocking_yaml_path):
        self._path = interlocking_yaml_path

    def resolve_train(self, action, inputs, state=None):
        return "3001-solo-match-complete"  # bare train_id, no structured model

    def execute(self, action, inputs, state=None, train_runner=TrainRunner):
        train_id = self.resolve_train(action, inputs, state)
        return train_runner(f"plan/_trains/{train_id}.yaml").execute(train_id, inputs)
